src/settings/SettingsModal.tsx

```diff
@@ -188,73 +211,64 @@ export const SettingsModal = (props: Props) => {
-        <ProfileSettingsTab client={props.client} />
-      </TabItem>
-    );
-  }
+  const developerTab = (
+    <TabItem
+      key="developer"
+      title={
+        <>
+          <DeveloperIcon width={16} height={16} />
+          <span>{t("Developer")}</span>
+        </>
+      }
+    >
+      <FieldRow>
+        <Body className={styles.fieldRowText}>
+          {t("Version: {{version}}", {
+            version: import.meta.env.VITE_APP_VERSION || "dev",
+          })}
+        </Body>
+      </FieldRow>
+      <FieldRow>
+        <InputField
+          id="showInspector"
+          name="inspector"
+          label={t("Show call inspector")}
+          type="checkbox"
+          checked={showInspector}
+          onChange={(e: ChangeEvent<HTMLInputElement>) =>
+            setShowInspector(e.target.checked)
+          }
+        />
+      </FieldRow>
+      <FieldRow>
+        <InputField
+          id="showConnectionStats"
+          name="connection-stats"
+          label={t("Show connection stats")}
+          type="checkbox"
+          checked={showConnectionStats}
+          onChange={(e: ChangeEvent<HTMLInputElement>) =>
+            setShowConnectionStats(e.target.checked)
+          }
+        />
+      </FieldRow>
+      <FieldRow>
+        <Button onPress={downloadDebugLog}>{t("Download debug logs")}</Button>
+      </FieldRow>
+    </TabItem>
+  );
 
-  if (developerSettingsTab) {
-    tabs.push(
-      <TabItem
-        key="developer"
-        title={
-          <>
-            <DeveloperIcon width={16} height={16} />
-            <span>{t("Developer")}</span>
-          </>
-        }
-      >
-        <FieldRow>
-          <Body className={styles.fieldRowText}>
-            {t("Version: {{version}}", {
-              version: import.meta.env.VITE_APP_VERSION || "dev",
-            })}
-          </Body>
-        </FieldRow>
-        <FieldRow>
-          <InputField
-            id="showInspector"
-            name="inspector"
-            label={t("Show call inspector")}
-            type="checkbox"
-            checked={showInspector}
-            onChange={(e: ChangeEvent<HTMLInputElement>) =>
-              setShowInspector(e.target.checked)
-            }
-          />
-        </FieldRow>
-        <FieldRow>
-          <InputField
-            id="showConnectionStats"
-            name="connection-stats"
-            label={t("Show connection stats")}
-            type="checkbox"
-            checked={showConnectionStats}
-            onChange={(e: ChangeEvent<HTMLInputElement>) =>
-              setShowConnectionStats(e.target.checked)
-            }
-          />
-        </FieldRow>
-        <FieldRow>
-          <Button onPress={downloadDebugLog}>{t("Download debug logs")}</Button>
-        </FieldRow>
-      </TabItem>
-    );
-  }
+  const tabs = [
+    audioTab,
+    videoTab,
+    ...(isEmbedded ? [] : [profileTab]),
+    feedbackTab,
+    moreTab,
+    ...(developerSettingsTab ? [developerTab] : []),
+  ];
```

Comment:
I think I'd prefer using `tabs.push()` the conditions here are somewhat weirdly written